### ISSUE TITLE
just-clone doesn't clone sets and maps

### ISSUE THREAD

**Issue by FarCryFromHuman** [angus-c/just#456]
This may be intended, but I discovered through some debugging that Sets I was using deep in my object structure were not being cloned; the references were being preserved.

Looking at the code, it's because there's no Set or Map handler in the algorithm. On line 24, where the type of the value currently being cloned is fetched, Set type objects return 'Set':
`line 24: var type = {}.toString.call(value).slice(8, -1);`

An example from Chrome:
```js
const set = new Set([1, 2, 3, 4])
> undefined
type = {}.toString.call(set).slice(8, -1)
> 'Set'
```
The fix is somewhat involved, as `result` needs to be assigned the correct object type, and then the 'add/set' functions need to be used, but not too bad.

**Comment by angus-c** [comment]
Yeah not yet supported. I'll try to get to it soon (unless you want to give it a go :))

**Comment by FarCryFromHuman** [comment]
I ended up putting together something fast and dirty that serves my purposes (managing sets and maps within immer drafts):
```js
/**
 * Deep copies the provided value - will copy functions by reference
 * WARNING: Only supports primitives, Sets, Maps, and standard objects and arrays
 * @param {any} value The value to deep copy
 * @returns
 */
export function clone(value) {
  // value is a function or a primitive, return value/function reference
  if (value instanceof Function || !(value instanceof Object)) return value;

  // value is an object, use interface type to determine how to process value
  let arr;
  let addFunc;
  let result;
  if (value instanceof Set) {
    arr = [...value];
    addFunc = (item, set) => set.add(item);
    result = new Set();
  } else if (value instanceof Map) {
    arr = [...value];
    addFunc = ([key, item], map) => map.set(key, item);
    result = new Map();
  } else if (value instanceof Array) {
    arr = value;
    addFunc = (item, array) => array.push(item);
    result = [];
  } else {
    // value must be standard object
    arr = Object.entries(value);
    addFunc = ([key, item], object) => (object[key] = item);
    result = {};
  }

  // iterate over object, recursively cloning over each item
  arr.forEach((item) => addFunc(clone(item), result));

  return result;
}
```

It's not great but it does the job for the moment. If I have time later on I'll put up a PR for you.

**Comment by angus-c** [comment]
Fixed in just-clone@5.1.0 (see #458) 
This should work for Sets/Maps as root object or as object properties
